server/src-lib/Hasura/GraphQL/Schema/Remote.hs

```diff
@@ -219,14 +258,24 @@ remoteSchemaUnion
   . (MonadSchema n m, MonadError QErr m)
   => SchemaIntrospection
   -> G.UnionTypeDefinition
-  -> m (Parser 'Output n ())
+  -> m (Parser 'Output n (SelectionSet NoFragments G.Name))
 remoteSchemaUnion schemaDoc defn@(G.UnionTypeDefinition description name _directives objectNames) =
   P.memoizeOn 'remoteSchemaObject defn do
-  objDefs <- traverse getObject objectNames
-  objs :: [Parser 'Output n ()] <- traverse (remoteSchemaObject schemaDoc) objDefs
+  objs :: [Parser 'Output n (Name, (SelectionSet NoFragments G.Name))] <-
+    traverse (\objName -> do
+                 obj <- getObject objName
+                 rsObj <- remoteSchemaObject schemaDoc obj
+                 return $ (objName,) <$> rsObj)
+             objectNames
```

Comment:
This code looks very similar to the code that's part of `remoteSchemaInterface`. Can we deduplicate this?

Reply:
Done